wgpu-types/src/lib.rs

```diff
@@ -1796,7 +1797,8 @@ impl<L> TextureViewDescriptor<L> {
     }
 
     pub fn array_layer_count(&mut self, array_layer_count: u32) -> &mut Self {
-        self.array_layer_count = Some(array_layer_count);
+        self.array_layer_count
+            .replace(NonZeroU32::new(array_layer_count).expect("Invalid array_layer_count"));
```

Comment:
nit: I think just `unwrap()` is fine. We can hardly expect anybody to pass `0` there, and it will be fairly clear why this fails